ledger/sync/src/ledger_sync/ledger_sync_service.rs

```diff
@@ -188,27 +188,20 @@ impl<L: Ledger, BC: BlockchainConnection + 'static, TF: TransactionsFetcher + 's
     }
 
     /// Append safe blocks to the local ledger.
-    fn append_safe_blocks(
-        &mut self,
-        blocks_and_contents: &[(Block, BlockContents)],
-    ) -> Result<(), LedgerSyncError> {
+    fn append_safe_blocks(&mut self, blocks: &[BlockData]) -> Result<(), LedgerSyncError> {
         log::info!(
             self.logger,
             "Appending {} blocks to ledger, which currently has {} blocks",
-            blocks_and_contents.len(),
+            blocks.len(),
             self.ledger
                 .num_blocks()
                 .expect("failed getting number of blocks"),
         );
-        mc_common::trace_time!(
-            self.logger,
-            "Appended {} blocks to ledger",
-            blocks_and_contents.len()
-        );
+        mc_common::trace_time!(self.logger, "Appended {} blocks to ledger", blocks.len());
 
-        for (block, contents) in blocks_and_contents {
+        for block_data in blocks {
             let append_block_start = SystemTime::now();
-            self.ledger.append_block(block, contents, None)?;
+            self.ledger.append_block_data(block_data)?;
```

Comment:
Perhaps we shouldn't append the signature here, and possibly not even the metadata; see discussion at https://github.com/mobilecoinfoundation/mcips/pull/43/files#r898367411

Reply:
#2216 handles this